Service/IndexService.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace EMS\CoreBundle\Service;
+
+use Elasticsearch\Client;
+use Psr\Log\LoggerInterface;
+
+final class IndexService
+{
+    /** @var AliasService */
+    private $aliasService;
+    /** @var Client */
+    private $elasticSearchClient;
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(AliasService $aliasService, Client $elasticSearchClient, LoggerInterface $logger)
+    {
+        $this->aliasService = $aliasService;
+        $this->elasticSearchClient = $elasticSearchClient;
+        $this->logger = $logger;
+    }
+
+    public function deleteOrphanIndexes(): void
+    {
+        $this->aliasService->build();
```

Comment:
Why is this build() call needed? In the old code it was never called in the indexController function deleteOrphansIndexesAction()?

Reply:
If I don't build it up front, getOrphanIndexes() returns an empty array.

Reply:
So this orphaned index action was never working before ?

Reply:
Nvm, it was "getElasticSearch()" that invoked the build.

I don't realy like this external call to ->build(), but that is out of scope for this change.